registry/store/eureka/client.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2016 IBM Corporation
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package eureka
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	log "github.com/Sirupsen/logrus"
+
+	eurekaapi "github.com/amalgam8/amalgam8/registry/api/protocol/eureka"
+	"github.com/amalgam8/amalgam8/registry/utils/logging"
+)
+
+const (
+	connectTimeout = time.Duration(2) * time.Second
+	requestTimeout = time.Duration(30) * time.Second
+)
+
+type eurekaClient struct {
```

Comment:
Have we considered using one of the open source Eureka clients?
Not that this is a huge file so it may be ok to avoid another external dependency.